vm/class.go

```diff
@@ -676,11 +676,13 @@ func builtinClassCommonInstanceMethods() []*BuiltinMethodObject {
 			},
 		},
 		{
-			// Inverts the boolean value.
+			// Inverts the boolean value. Any objects other than `nil` and `false` are `true`, thus returns `false`.
 			//
 			// ```ruby
 			// !true  # => false
 			// !false # => true
+			// !nil   # => true
```

Comment:
Do we have test case for this condition? According to the code below it looks like this will return false because it's not a boolean object

Reply:
Added evaluation tests for that.
`!nil` works as expected to be `true`: the negation of `false` that is equivalent to`nil`